### ISSUE TITLE
Game crash while farming, planting threes

### ISSUE THREAD

**Issue by itenev** [abearcodes/Valheim#19]
Steps to reproduce:

- Plant some turnip/carrot seeds.
- Planting some Beech seeds.

Observed behavior:

Game crash. The following errors (multiple instances) are observed in LogOutput.log:  
[Info   : Unity Log] 03/06/2021 10:35:02: Placed sapling_carrot

[Error  :Crafting with Containers] Invalid state reached! You might want to report this to the mod developer.
When removing 1 of $item_carrotseeds, amount of resources left to remove was still 1
Containers: 0. Players: 1.
Removed 1 "Carrot seeds". Touched 0 inventories
...
[Info   : Unity Log] 03/06/2021 10:39:49: Placed Beech_Sapling

[Error  :Crafting with Containers] Invalid state reached! You might want to report this to the mod developer.
When removing 1 of $item_beechseeds, amount of resources left to remove was still 1
Containers: 0. Players: 1.
Removed 1 "Beech seeds". Touched 0 inventories

Attached please find complete LogOutput.log
[LogOutput.log.zip](https://github.com/abearcodes/Valheim/files/6095074/LogOutput.log.zip)

**Comment by abearcodes** [comment]
Hello. Thank you for your report! This is a bug with the error reporting code that reports errors when it only took items from the players inventory and didn't touch any of the tests. The mod (and game) should work perfectly otherwise. 

Apart from just seeing the message did you experience any issues?

**Comment by abearcodes** [comment]
Yep, just reconfirmed this. Fix already implemented. Will be released in 0.0.8

**Comment by itenev** [comment]
Actually, yes. My game crashed for the first time, not sure if this is the reason tho. I am attaching my Player.log as well.

One thing I notice here is that the number of reported Players in the message grows from 1 to 4 immediately before the crash and this is single player, not a server:

Containers: 0. Players: 4. 

[Player.log.zip](https://github.com/abearcodes/Valheim/files/6095190/Player.log.zip)

**Comment by abearcodes** [comment]
The players counter would've been a good indicator. But I just checked and.. well. I did a bad job coding the player indicator. It searches for "Character"s, but all animals are also considered a character... So while that indicator is not "wrong" per se, the label is. It's actually showing the amount of all character-like entities around. I fixed the indicator going forward, but this doesn't really help us with your crash...

As for this crash, you are running with Vulkan on Linux. I myself have had some issues from time to time with some Unity games on my Ubuntu setup, including crashes.  

[Valheim also seems crashy in general.](https://valheimbugs.featureupvote.com/suggestions/158466/fix-the-crashing) I checked the Player.log you provided for more details, and while my knowledge of the inner workings of Unity, Linux, and Vulkan are nowhere near enough to properly assess this, my best guess is that's it has something to do with the Vulkan implementation on Linux. 

At this point I don't think this is related to Crafting With Containers or other mods (unless they do some gpu-level stuff via native libraries). 

I will release a new version that fixes the false positives in the console output, so thank you for alerting me on this. I will close this ticket once the logging issues are released.
